### PR TITLE
CharityService

### PR DESCRIPTION
**Pull request by rubenev**
Package updates:
"Microsoft.NET.Test.Sdk" Version="15.7.0"
"MSTest.TestAdapter" Version="1.2.1"
"MSTest.TestFramework" Version="1.2.1"

Only thing missing is an complete dataset for charity. momentairly working with only 3 variables:
CharityID
NameCharity
NameOwner

After research is done CharityEntity can be completed

### CODE REVIEW

**Comment by rubenev** [comment]
Not sure how to fix these conflicts. shouldnt master just get on its knees and take my updated packages?

File: Sombra.CharityService/DAL/CharityContext.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Sombra.CharityService.DAL
+{
+    public class CharityContext : SombraContext<CharityContext>
+    {
+        public CharityContext() { }
+        public CharityContext(DbContextOptions<CharityContext> options) : base(options)
+        {
+        }
+        public DbSet<CharityEntity> Charity { get; set; }
```
Comment (JelleKerkstra) [inline Sombra.CharityService/DAL/CharityContext.cs]: *Charities (Het is meervoud want er zit een tabel met charities achter)

File: Sombra.CharityService/DAL/CharityEntity.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Sombra.CharityService.DAL
+{
+    public class CharityEntity : Entity
+    {
+        // TODO at more data relevant for charity
+        public string CharityId { get; set; }
```
Comment (JelleKerkstra) [inline Sombra.CharityService/DAL/CharityEntity.cs]: public Guid CharityKey

File: Sombra.Messaging/Requests/CharityRequest.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+using Sombra.Messaging.Responses;
+
+namespace Sombra.Messaging.Requests
+{
+    public class CharityRequest : Request<CharityResponse>
```
Comment (JelleKerkstra) [inline Sombra.Messaging/Requests/CharityRequest.cs]: CreateCharityRequest

File: Sombra.Messaging/Responses/CharityResponse.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Collections.Generic;
+
+namespace Sombra.Messaging.Responses{
+    public class CharityResponse : Response {
```
Comment (JelleKerkstra) [inline Sombra.Messaging/Responses/CharityResponse.cs]: CreateCharityResponse

File: Sombra.CharityService/GetCharityByKeyRequestHandler.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            _charityeContext = charityContext;
+        }
+
+        public async Task<CharityResponse> Handle(CharityRequest message)
+        {
+            var response = new CharityResponse(true);
+
+            var charity = await _charityeContext.Charity.Where(b => b.CharityId.Equals(message.CharityId)).Select(a => a).FirstOrDefaultAsync();
+            if (charity == null) return new CharityResponse(false);
```
Comment (JelleKerkstra) [inline Sombra.CharityService/GetCharityByKeyRequestHandler.cs]: CharityResponse() 
Als je een bool geen waarde geeft, is ie standaard false
default(bool) == false

File: Sombra.CharityService/appsettings.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+﻿{
+  "RABBITMQ_CONNECTIONSTRING": "host=localhost",
+  "TEMPLATE_DB_CONNECTIONSTRING": "Server=.\\SQLEXPRESS;Database=Charitys;Trusted_Connection=True;"
```
Comment (JelleKerkstra) [inline Sombra.CharityService/appsettings.json]: CHARITY_DB_CONNECTIONSTRING

**Pull request merged**